lib/practice_exercises.rb

```diff
@@ -1,13 +1,46 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: 0(1)
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  string = string.downcase
+  if string.length == 1 || string == ""
+    return true
+  elsif string.length == 0 || string == nil
+    raise StandardError.new ("Not a Palindrome")
+  end
+  i = string[0]
+  j = string[-1]
+  until i == j
+    if string[i] == string[j]
+      i += 1
+      j -= -1
+    else 
+      return false
+    end
+    return true
+  end
+  return true
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(NM log M)
+# Space Complexity: O(1)
+# vertical scanning method:
+# https://medium.com/@d_dchris/10-methods-to-solve-the-longest-common-prefix-problem-using-python-leetcode-14-a87bb3eb0f3a
 def longest_prefix(strings)
```

Comment:
There's nothing here that you're cutting in half with each iteration of the loop, instead you're looping through each letter in each word so I would say this is `O(n * m)`.

Otherwise this works!  👍